.circleci/config.yml

```diff
@@ -198,73 +150,20 @@ jobs:
           paths:
             - ~/.ccache
 
-  build-release-binary-osx:
-    executor: macos
-    steps:
-      - install_macos_dependencies
-      - checkout
-      - run: ./scripts/format.sh && ./scripts/error_on_dirty.sh
-      - run: git submodule sync && git submodule update --init
-      - restore_cache:
-          keys:
-            - ccache-release-macos-{{ .Branch }}
-            - ccache-release-macos
-      - run: mkdir -p build
-      - run: cd build && cmake .. -DBUILD_SHARED_LIBS=Off -DENABLE_PYTHON_BINDINGS=Off -DENABLE_DATA_TOOLS=OFF -DENABLE_SERVICES=OFF -DBoost_USE_STATIC_LIBS=ON -DProtobuf_USE_STATIC_LIBS=ON -DLZ4_USE_STATIC_LIBS=ON
-      - run: make -C build -j4
-      - save_cache:
-          key: ccache-release-macos-{{ .Branch }}-{{ epoch }}
-          paths:
-            - ~/.ccache
-
-  build-debug-binary-osx:
-    executor: macos
-    steps:
-      - install_macos_dependencies
-      - checkout
-      - run: ./scripts/format.sh && ./scripts/error_on_dirty.sh
-      - run: git submodule sync && git submodule update --init
-      - restore_cache:
-          keys:
-            - ccache-debug-macos-{{ .Branch }}
-            - ccache-debug-macos
-      - run: mkdir -p build
-      - run: cd build && cmake .. -DCMAKE_BUILD_TYPE=Debug -DENABLE_PYTHON_BINDINGS=Off -DBUILD_SHARED_LIBS=Off -DENABLE_DATA_TOOLS=OFF -DENABLE_SERVICES=OFF -DBoost_USE_STATIC_LIBS=ON -DProtobuf_USE_STATIC_LIBS=ON -DLZ4_USE_STATIC_LIBS=ON
-      - run: make -C build -j4
-      - save_cache:
-          key: ccache-debug-macos-{{ .Branch }}-{{ epoch }}
-          paths:
-            - ~/.ccache
-
 workflows:
   version: 2
   build_test_publish:
     jobs:
       - build-docker-images
       - lint-build-debug
-      - build-release
-      - build-osx
-      - build-release-binary-osx:
           filters:
             tags:
-              only: /.*/
-            branches:
               ignore: /.*/
-      - build-debug-binary-osx:
-          filters:
-            tags:
-              only: /.*/
-            branches:
-              ignore: /.*/
-      - build-release-binary-linux:
+      - build-release
           filters:
             tags:
-              only: /.*/
-            branches:
               ignore: /.*/
-      - build-debug-binary-linux:
+      - build-osx
           filters:
             tags:
-              only: /.*/
-            branches:
               ignore: /.*/
```

Comment:
this part of the diff is hard to read but basically this is what i was seeing. when i tag the repo and it kicks off a build, because we had `filters.tags.only = .*` these were the only builds that got run on the tag. which means that when tagged the actual docker build was not run! see here:

![image](https://user-images.githubusercontent.com/697548/115616584-5b0d0d00-a2be-11eb-99a7-cff522056934.png)

it only ran those, which are useless to everyone. so ive changed it here so that of the builds that we care about (linux debug, linux release, docker and macos) tags will skip all of them execpt the docker build because the point of our tagging is basically to create runnable/usable artifacts rather than check that the code is working (presumably we do that by merging to master before tagging :smile: )